src/types/index.ts

```diff
@@ -382,19 +382,9 @@ export class StarknetContractFactory {
         }
 
         const executedOutput = executed.stdout.toString();
-        const classHash = extractClassHash(executedOutput);
         const txHash = extractTxHash(executedOutput);
 
-        return new Promise((resolve, reject) => {
-            iterativelyCheckStatus(
-                txHash,
-                this.hre.starknetWrapper,
-                () => resolve(classHash),
```

Comment:
My idea was to keep on iteratively checking status, but eventually resolving not with `classHash` but with `txHash`, since the class hash can be obtained in different ways (`contractFactory.getClassHash()` as mentioned in #313).

Your change would make declare txs the only ones which don't wait for the transaction to become accepted. Ideally all transactions would behave the same (or is there an advantage in treating declaration differently?)

Reply:
Fair point. Consistency is king. This will be changed👍

Reply:
Re-introduced iterative checking in  3662a4435bca6a25e6c0193d87dd5d5a45474c27.